microsoft-edge/webview2/how-to/prerelease-testing.md

```diff
@@ -0,0 +1,190 @@
+
+To catch app-specific regressions before WebView2 changes ship in Edge Stable, test your particular app against the prerelease channels of Microsoft Edge.  WebView2 is tested in a variety of general scenarios, but it's possible that some of the specific scenarios of your app are not covered.
+
+
+<!-- ====================================================================== -->
+## Start by doing automated testing against Edge Canary
+
+Start with automated testing against the Microsoft Edge **Canary** channel, which ships daily.  This enables you to catch issues as early as possible. The Microsoft Edge **Dev** channel ships weekly, and is also a good option.
+
+Do the following steps:
+
+1. Set the preview channel for your app.  See [Set the preview channel by using ChannelSearchKind](#set-the-preview-channel-by-using-channelsearchkind), below.
+
+1. Deploy the non-stable channels.  See [How to deploy non-stable channels](#how-to-deploy-non-stable-channels), below.
+
+1. Conduct automated testing on your app running against the non-stable channels.  See [How to do automated testing](#how-to-do-automated-testing), below.
+
+1. Compare the results with the baseline.  See [The baseline release of WebView2 for testing](#the-baseline-release-of-webview2-for-testing), below.
+
+Upon finding issues, you can report them via the [WebView2 feedback repo](https://github.com/MicrosoftEdge/WebView2Feedback).
+
+
+<!-- ====================================================================== -->
+## Setting the preview channel by using ChannelSearchKind to set the channel search order
+
+The channel search kind controls the channel search order.  By default, the WebView2 loader searches for binaries from most-stable to least-stable (Canary), using the first binary that's found:
+
+```
+WebView2 Runtime (Stable) ->  Edge Beta -> Edge Dev -> Edge Canary
+```
+
+To do prerelease testing, reverse the channel search order, so that the loader looks for binaries in order from least-stable (Canary) to most-stable:
+
+```
+Edge Canary -> Edge Dev -> Edge Beta -> WebView2 Runtime (Stable)
+```
+
+The channel search kind can be set per-app through a registry key, an environment variable, an API, or a group policy.
+
+The value `1` indicates reversed search order; that is, from least-stable (Edge Canary) to most-stable.
+
+
+##### [Registry key](#tab/registry-key)
+
+```reg
+REG ADD <HKLM/HKCU>\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind /v WebView2APISample.exe /t REG_DWORD /d 1
+```
+
+Replace `WebView2APISample.exe` with your own app executable name or the application user model ID.
+
+
+##### [Environment variable](#tab/environment-variable)
+
+Name: `WEBVIEW2_CHANNEL_SEARCH_KIND`  
+Value: `1`
+
+Note that the environment variable will be applied to all apps that use WebView2 on the machine, unlike the registry key, which can be set per app.<!-- todo: confirm this line, copied from self-hosting.md -->
```

Comment:
"Note that the environment variable will be applied to all apps that use WebView2 on the machine, unlike the registry key, which can be set per app."
Confirm that this line should be added here.  Copied it from self-hosting.md.
@victorhuangwq